python/lsst/daf/butler/tests/_examplePythonTypes.py

```diff
@@ -237,18 +251,20 @@ def makeFromDict(cls, exportDict):
         newobject : `MetricsExample`
             New `MetricsExample` object.
         """
-        data = None
-        if "data" in exportDict:
-            data = exportDict["data"]
+        data = exportDict["data"] if "data" in exportDict else None
+        assert data is None or isinstance(data, list)
```

Comment:
These days you should be able to write:

    assert isinstance(data, list | None)

Reply:
I had no idea that syntax worked (and thought at least I needed `NoneType` for `isinstance`).

Reply:
PEP 604 says it's OK to use just None.

Reply:
```python
>>> x = None
>>> isinstance(x, None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union
>>> isinstance(x, types.NoneType)
True
>>> isinstance("str", types.NoneType)
False
>>> isinstance(x, str | None)
True
>>> isinstance("x", str | None)
True
>>> isinstance(5, str | None)
False
```

So @andy-slac is correct and I am very surprised indeed. It turns out it only works if you have two types.

Reply:
A very funny development is that whilst python is absolutely fine with the syntax, mypy 0.991 requires me to use `types.NoneType` (but does still allow the `str | types.NoneType` syntax).

Reply:
Sorry. I was unclear. I meant `isinstance(None, None)` does not work but `isinstance(None, None | a | b | ...)` does work (so 2 or more).